### PR TITLE
[CARBONDATA-3889] Cleanup code in carbondata-core module

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
1. Redundant 'if' statement
2. Array access many times in for loop
3. Repeated switch branch
4. Use StringBuffer
 
 ### What changes were proposed in this PR?
1. Change redundant 'if' statement
2. Use enhanced for loop
3. Merge switch branch
4. Use StringBuilder instead of StringBuffer
    
 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1787/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3528/

File: core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java [hunk 1]
```diff
@@ -283,8 +283,8 @@ public BigDecimal getDecimal(int rowId) {
   @Override
   public byte[][] getByteArrayPage() {
     byte[][] data = new byte[arrayElementCount][];
-    for (int i = 0; i < arrayElementCount; i++) {
-      data[i] = fixedLengthData[i];
+    if (arrayElementCount >= 0) {
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java:286]:
```suggestion
    if (arrayElementCount = 0) {
```

File: core/src/main/java/org/apache/carbondata/core/index/IndexMeta.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public String toString() {
     return new StringBuilder("IndexMeta{")
         .append("indexName='").append(indexName).append('\'')
         .append(", indexedColumns=[")
-        .append(StringUtils.join(getIndexedColumnNames(), ", ")).append("]\'")
+        .append(StringUtils.join(getIndexedColumnNames(), ", ")).append("]'")
         .append(", optimizedOperation=").append(optimizedOperation)
         .append('}')
         .toString();
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/index/IndexMeta.java:84]:
```
    return new StringBuilder("IndexMeta{indexName='")
        .append(indexName)
        .append("', indexedColumns='[")
        .append(StringUtils.join(getIndexedColumnNames(), ", "))
        .append("]', optimizedOperation='")
        .append(optimizedOperation)
        .append("'}")
        .toString();
```
keep same quotation marks

File: core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java [hunk 3]
```diff
@@ -105,20 +106,18 @@ public boolean equals(Object o) {
...
+    return Objects.equals(blockletId, blocklet.blockletId);
   }
 
   @Override
   public String toString() {
-    final StringBuffer sb = new StringBuffer("Blocklet{");
+    final StringBuilder sb = new StringBuilder("Blocklet{");
     sb.append("filePath='").append(filePath).append('\'');
     sb.append(", blockletId='").append(blockletId).append('\'');
     sb.append('}');
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java:123]: ditto

File: core/src/main/java/org/apache/carbondata/core/metadata/datatype/Field.java [hunk 4]
```diff
@@ -193,6 +192,7 @@ public void setParent(String parent) {
   }
 
   public String getStoreType() {
+    String storeType = "columnar";
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/metadata/datatype/Field.java:195]: Q1: why this change? why not mark the string as static instead?
Q2: this method is never used

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java [hunk 5]
```diff
@@ -722,15 +719,11 @@ public int getBlockletSizeInMB() {
...
         .get(CarbonCommonConstants.BUCKET_HASH_METHOD);
-    if (configuredMethod == null) {
-      return CarbonCommonConstants.BUCKET_HASH_METHOD_DEFAULT;
-    } else {
-      if (CarbonCommonConstants.BUCKET_HASH_METHOD_NATIVE.equals(configuredMethod)) {
-        return CarbonCommonConstants.BUCKET_HASH_METHOD_NATIVE;
-      }
-      // by default we use spark_hash_expression hash method
-      return CarbonCommonConstants.BUCKET_HASH_METHOD_DEFAULT;
+    if (CarbonCommonConstants.BUCKET_HASH_METHOD_NATIVE.equals(configuredMethod)) {
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java:722]: should keep (not) null check

File: core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java (hunk 1 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java:286]: change to "if (arrayElementCount = 0) {"

File: core/src/main/java/org/apache/carbondata/core/index/IndexMeta.java (hunk 2 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/index/IndexMeta.java]: done

File: core/src/main/java/org/apache/carbondata/core/metadata/datatype/Field.java (hunk 4 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/metadata/datatype/Field.java]: removed this method

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java (hunk 5 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java:722]: "==" and "instanceof" contain null check

File: core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java (hunk 3 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java]: done

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java (hunk 5 above)
Comment (kevinjmh) [reply core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java:722]: ok

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3553/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1813/

**Comment by kevinjmh** [comment]
LGTM
